Clean up the LazyTensor class

desertfire opened the pull request:
- Move aten-op methods out
- Move distributed-related methods out
- Move compiler/execution/synchronization-related methods out

desertfire committed:
LazyTensor Cleanup - Move static aten op methods out

desertfire committed:
LazyTensor Cleanup - Move distributed-related methods out

desertfire committed:
LazyTensor Cleanup - Move class DeviceContextArena out

JackCaoG commented:
@wconstab I remembered that you mentioned you will try to codegen pretty most of the stuff under today's `tensor_method.cpp`, is that still true?

wconstab commented:
@JackCaoG 
> I remembered that you mentioned you will try to codegen pretty most of the stuff under today's `tensor_method.cpp`, is that still true?

Yes and no.  In the new design, we simplified things a bit so the NativeFunction itself directly constructs the IR node, there isn't really a separate method that is equivalent to existing LazyTensor::add.  But we do codegen all of what is needed, so, for the TS backend at least, we'd plan over time to delete the tensor_methods.cpp file after we switch all the ops to codegen.

desertfire committed:
LazyTensor Cleanup - Move class DeviceLockerArena out

desertfire committed:
LazyTensor Cleanup - Move class DataCacheArena out

desertfire committed:
LazyTensor Cleanup - Move graph compiler/execution part out

Summary: Move the following public methods and relevant private methods into
LazyGraphExecutor
- GetLiveTensors
- SyncLiveTensorsGraph
- SyncTensorsGraph
- MarkStep
- WaitDeviceOps
- GetTensors

wconstab reviewed (approved):
nice cleanup!
